adafruit_am2320.py

```diff
@@ -110,17 +110,36 @@ def __init__(self, i2c_bus: I2C, address: int = AM2320_DEFAULT_ADDR):
 
     def _read_register(self, register: int, length: int) -> bytearray:
         with self._i2c as i2c:
+            write_success = False
             # wake up sensor
-            try:
-                i2c.write(bytes([0x00]))
-            except OSError:
-                pass
+            for _ in range(3):
+                try:
+                    i2c.write(bytes([0x00]))
+                    write_success = True
+                    time.sleep(0.1)  # wait 100ms
+                    break
+                except OSError:
+                    write_success = False
```

Comment:
`write_success` can already only be `False` in the `except` arm, so setting it is redundant.
```suggestion
                    pass
```